Zend/zend_attributes.c

```diff
@@ -84,6 +84,9 @@ static void validate_allow_dynamic_properties(
 			ZSTR_VAL(scope->name)
 		);
 	}
+	if (scope->ce_flags & ZEND_ACC_ENUM) {
+		zend_error_noreturn(E_ERROR, "Cannot apply #[AllowDynamicProperties] to enum");
```

Comment:
I think you can add the component id here (like other cases) wdyt ?

Reply:
I was just copying from the handling of traits and interfaces, but yeah, good point, done. It would probably be useful to add the same component id to traits' and interfaces' error messages, I'll send a patch for that (but targeting `master` since it isn't a bug fix)

Reply:
nice ;-) cc @iluuu1994